src/Spies/GlobalSpies.php

```diff
@@ -121,10 +131,7 @@ private static function replace_global_function( $function_name ) {
 		}
 		self::$redefined_functions[ $function_name ] = \Patchwork\redefine( $function_name, function() use ( $function_name ) {
 			$value = \Spies\GlobalSpies::handle_call_for( $function_name, func_get_args(), [ 'return_falsey_objects' => true ] );
-			if ( isset( $value ) ) {
-				return self::filter_return_for( $value );
-			}
-			return \Patchwork\relay();
```

Comment:
I am not entirely sure why we need to relay here.
I did not intend to change this part (this is not directly part of the fix) but without this the call count of a stubbed method gets increased. For example:

```
1) ConsecutiveExecutionTest::test_undefined_function_can_be_restubbed
Failed asserting that 2 matches expected 1.

spies/tests/ConsecutiveExecutionTest.php:23
```

Reply:
Ok, I followed the code and it looks like when `GlobalSpies::handle_call_for()` is called, it eventually calls `GlobalSpies::call_original_global_function()` thanks to `Spy->get_return_for()`, which uses `\Patchwork\restore()` to call the function before recreating the mock with `GlobalSpies::replace_global_function()`.

That explains why this works. Maybe that wasn't always the case which would explain the relay? Or maybe there's another edge case we're not considering here. 🤷